cmake/FindShapefile.cmake

```diff
@@ -0,0 +1,17 @@
+# - Try to find Shapefile C Library
+#   http://shapelib.maptools.org/
+#
+# Once done, this will define
+#
+#  Shapefile_FOUND - system has Shapefile
+#  Shapefile_INCLUDE_DIRS - the Shapefile include directories
+#  Shapefile_LIBRARIES - link these to use Shapefile
+
+find_path(Shapefile_INCLUDE_DIR NAMES shapefil.h)
+find_library(Shapefile_LIBRARY NAMES shp)
+
+set(Shapefile_FOUND TRUE)
+if ("${Shapefile_INCLUDE_DIR}" STREQUAL "Shapefile_INCLUDE_DIR-NOTFOUND" OR
+    "${Shapefile_LIBRARY}" STREQUAL "Shapefile_LIBRARY-NOTFOUND")
+  set(Shapefile_FOUND FALSE)
+endif()
```

Comment:
I think the canonical way to do this is to dispatch to CMake's FindPackageHandleStandardArgs module, e.g.

https://github.com/daniel-j-h/cku/blob/master/cmake/FindCapstone.cmake